### PR TITLE
[#9435] Prevent multiaccount duplication

### LINKED ISSUE #9435: it must not be possible to create two accounts (one with keycard, one without keycard) with the same seed
**Description** 
It is currently possible to create two accounts that have the same mnemonic (hence the same seed) one with keycard, one without keycard. 

**Problem**
This is a security problem and totally defeats the purpose of keycard. The problem is that it gives a false sense of security to the user
- that uses keycard to keep his private keys offline, in a cold storage
- while they are still stored on the phone 

**Reproduction** 
-  create a keycard account: generate a key > select storage: keycard. At this stage write down the mnemonic.
- create a non-keycard account with the same seed: access key > enter seed phrase 

reproduced in #9062 

**Solution**
whenever a new status account (keycard or non-keycard) is created we must make sure that there is no existing account with the same seed on the phone (keycard or non-keycard)

cc @rachelhamlin @corpetty @rasom

### PR DESCRIPTION
**Pull request by rasom**
fix #9435 

Currently we have two ways to restore a multiaccount:
- by entering a mnemonic phrase
- by pairing a keycard with an existing multiaccount

In both cases, when we detect that a user tries to recover an existing
multiaccount we interrupt recovering and propose them to unlock that
multiaccount instead.

note:
Replacement of master key `address` with `key-uid` as a multiaccount id didn't happen in this PR because it will require changes in many places in the app, thus this change can cause many regressions. Current PR is already big enough and solves its objective.

status: ready

### CODE REVIEW

File: src/status_im/hardwallet/core.cljs [hunk 1]
```diff
@@ -1752,7 +1753,7 @@
                 :address              address
                 :publicKey            public-key
                 :keycard-instance-uid instance-uid
-                :keycard-key-uid      key-uid
+                :keyUid               (ethereum/normalized-hex key-uid)
```
Comment (flexsurfer) [inline src/status_im/hardwallet/core.cljs:1758]: why camelCase?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/hardwallet/core.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/hardwallet/core.cljs:1758]: It comes that way from go side, I didn't want to mix cases

**Comment by churik** [comment]
@rasom 
failed e2e are not related to PR - some of them are fixed already and several are existing issues.


#### ISSUE 1. Can't login if logged out (your PR) - "Looks you've tapped wrong keycard" error

Steps:
- create account in PR
- log out
- try to login

Expected result:
can login

Actual result:
 "Looks you've tapped wrong keycard"
Full logcat: [wrong-keycard.log](https://github.com/status-im/status-react/files/3893163/wrong-keycard.log)
The issue does not exist in nightly 25/12/2019 (last successful, build 1336)

**Comment by rasom** [comment]
@churik issue 1 should be fixed now

**Comment by churik** [comment]
@rasom 
Issue 1 is fixed.

### Issue 2: can't unlock multiaccount due to error


Steps:
- create multiaccount in PR 
- backup seed phrase
- log out
- try to restore from account from seed phrase
- when the popup appears, select unlock

Expected result:
can login with keycard

Actual result:
 Error
```
1-27 19:33:57.945 21914 21944 I ReactNativeJS: Error: Please shake the phone to report this error and restart the app. multiaccount is currently empty, which means something went wrong when trying to update settings
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3281133)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3280418)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3224941)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at Eu (address at index.android.bundle:1:2115967)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:2994010)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:2977118)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3007516)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3224835)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3324827)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3323764)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3224941)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at Eu (address at index.android.bundle:1:2115967)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:2994010)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:2977118)
11-27 19:33:57.945 21914 21944 I ReactNativeJS:     at anonymous (address at index.android.bundle:1:3007516)
```

Full logcat: 
[unlock-log.log](https://github.com/status-im/status-react/files/3896680/unlock-log.log)

**Comment by rasom** [comment]
@churik 
issue 2 should be fixed now

**Comment by churik** [comment]
Tested on Android 8 (LG V20) and Android 9.2 (Xiaomi Mi Note 9 Pro):
- create account with KK > trying to restore the same on the same device with seed phrase > Unlock account

- create account with KK > trying to restore the same on the same device with seed phrase > Cancel > Login\logout

- create account with KK > trying to restore the same on the same device using KK> Unlock > Login\logout

- create account with KK + another account> trying to restore the same on the same device with seed phrase > Cancel > Login\logout

- create account > trying to restore the same multiaccount on the same device using KK > Unlock > Login\logout

- pair KK on account > trying to restore the same on the same device using seed phrase > Unlock > Login\logout

- port account to new KK >  trying to restore the same on the same device using seed phrase > Unlock > Login\logout

- several accounts >  trying to restore the same on the same device using seed phrase/KK > unlock one of them.

**Comment by churik** [comment]
@rasom please resolve the conflicts and tell me if retesting is required.
Thanks!

**Pull request merged**